extensions/iceberg/src/main/java/io/deephaven/iceberg/base/IcebergUtils.java

```diff
@@ -44,36 +49,18 @@ public final class IcebergUtils {
      *
      * @return A stream of {@link DataFile} objects.
      */
-    public static Stream<DataFile> getAllDataFiles(
-            @NotNull final Table table,
-            @NotNull final Snapshot snapshot) {
+    public static Stream<DataFile> allDataFiles(@NotNull final Table table, @NotNull final Snapshot snapshot) {
         final FileIO fileIO = table.io();
         try {
-            // Retrieve the manifest files from the snapshot
-            final List<ManifestFile> manifestFiles = snapshot.allManifests(fileIO);
-            return manifestFiles.stream()
-                    .peek(manifestFile -> {
-                        if (manifestFile.content() != ManifestContent.DATA) {
-                            throw new TableDataException(
-                                    String.format(
-                                            "%s:%d - only DATA manifest files are currently supported, encountered %s",
-                                            table, snapshot.snapshotId(), manifestFile.content()));
-                        }
-                    })
-                    .flatMap(manifestFile -> {
+            return allManifests(table, snapshot).stream()
+                    .map(manifestFile -> ManifestFiles.read(manifestFile, fileIO))
+                    .flatMap(manifestReader -> {
                         try {
-                            final ManifestReader<DataFile> reader = ManifestFiles.read(manifestFile, fileIO);
-                            return StreamSupport.stream(reader.spliterator(), false).onClose(() -> {
-                                try {
-                                    reader.close();
-                                } catch (IOException e) {
-                                    throw new UncheckedIOException(e);
-                                }
-                            });
+                            return toStream(manifestReader);
```

Comment:
A try / catch around toStream doesn't make any sense to me _unless_ we expect that `manifestReader.spliterator()` is doing real work, which I don't think it should be doing...